src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java

```diff
@@ -384,8 +393,17 @@ private void onSearchForRelationsSucceed(BibEntry entry, CheckListView<CitationR
                                              ObservableList<CitationRelationItem> observableList) {
         hideNodes(abortButton, progress);
 
-        observableList.setAll(fetchedList.stream().map(entr -> new CitationRelationItem(entr, false))
-                                         .collect(Collectors.toList()));
+        observableList.setAll(fetchedList.stream().map(entr -> {
+            Optional<BibEntry> duplicate = duplicateCheck.containsDuplicate(
+                    databaseContext.getDatabase(),
+                    entr,
+                    BibDatabaseModeDetection.inferMode(databaseContext.getDatabase()));
+
+            if (duplicate.isPresent()) {
+                return new CitationRelationItem(entr, duplicate.get(), true);
+            }
+            return new CitationRelationItem(entr, null, false);
```

Comment:
Not 100% sure of the concrete syntax, but you should get the idea!

```suggestion
            return duplicate.map(duplicate -> new CitationRelationItem(entr, duplicate, true)).orElse(new CitationRelationItem(entr, null, false));
```

You could even get rid of the variable "duplicate".